pkg/auth/ldaputil/testclient/testclient.go

```diff
@@ -0,0 +1,133 @@
+package testclient
+
+import (
+	"crypto/tls"
+
+	"github.com/go-ldap/ldap"
+	"github.com/openshift/origin/pkg/auth/ldaputil/ldapclient"
+)
+
+// Fake is a mock client for an LDAP server
+// The following methods define safe defaults for the return values. In order to adapt this test client
+// for a specific test, subtype it and override the method being tested.
+type Fake struct {
+	simpleBindResponse     *ldap.SimpleBindResult
```

Comment:
is there a good reason for these to private?  private parts of mock objects usually cause me pain.